src/components/markdown-renderer/basic-markdown-renderer.tsx

```diff
@@ -26,7 +15,7 @@ import { calculateNewLineNumberMapping } from './utils/line-number-mapping'
 
 export interface BasicMarkdownRendererProps {
   componentReplacers?: ComponentReplacer[],
-  onConfigureMarkdownIt?: (md: MarkdownIt) => void,
+  markdownIt?: MarkdownIt,
```

Comment:
IMHO the markdown it prop shouldn't be optional.

Reply:
But it needs to be. When I want to use a BasicMarkdownRender by itself it needs to configure its own markdownit

Reply:
The react way to do this would be to create a "MarkdownRenderer" component that takes a markdown it prop and a "BasicMarkdownRenderer" that configures a plain markdown it instance and passes it into a "MarkdownRenderer".

Reply:
Also: I don't see that BasicMarkdownRenderer is used without the prop.

Reply:
it should be in the help dialog.

Reply:
Instead of making the prop optional I would introduce a "plain markdown renderer" that uses the basicmarkdownrenderer

Reply:
On the one hand, I understand that another renderer would be better. But remember with the view-only mode I'm gonna separate the full-renderer once again into full- and scrollable-full-renderer. Anytime soon, we'll have more markdown-renderer-components than other components...

Reply:
So what? That is the purpose of a component system. To split up the tasks. You should use composition.

Reply:
And "there will be more renderer than other components" is a bit exaggerated 🤨. There will be the basic and this is used by plain, full (which is used by full-scrollable)

Reply:
Composition is one thing and I'm okay with it, if you do it the way you just detailed.
I just want to remind that during software development one should be careful about the maintainability. For new contributors, especially ones without greater React knowledge, it might seem too complex to work on these parts of the software.

Reply:
IMHO it's more confusing that the basic renderer takes an markdown it instance or not